integration/util.py

```diff
@@ -562,11 +603,27 @@ def outReceived(self, data):
             print("Saw '{}' in the logs".format(self._magic_text))
             self.magic_seen.callback(self)
 
-    def errReceived(self, data):
+    def err_received(self, data):
+        """
+        Called when non-JSON lines are received on stderr.
+
+        On Windows we use stderr for eliot logs from magic-folder.
+        But neither magic-folder nor tahoe guarantee that there is
+        no other output there, so we treat it as expected.
+        """
         with self._action.context():
             Message.log(message_type=u"err-received", data=data)
             sys.stdout.write(data)
 
+    def eliot_garbage_received(self, data):
+        """
+        Called when non-JSON lines are received on FD 3.
+
+        Since FD 3 is suppposed to only have eliot-logs, log them as malformed.
+        """
+        with self._action.context():
+            Message.log(message_type=u"malformed-eliot-log", data=data)
```

Comment:
Can we also make this into a test-failing event so that if/when such things arise in the future we can catch them quickly?  Or is there so much garbage on FD 3 that it's not even worth trying to make it clean?

Reply:
It is not straight-forward to do this for a couple of reasons:
- the service is not scoped to a single test
- I don't think raising an exception here would cause the test to fail, as it will be swallowed by the reactor.
I filed #510 to address this. As it stands, I don't think we should ever hit this, as FD 3 is explicitly for eliot log messages.